embassy-stm32f4/src/can.rs

```diff
@@ -78,3 +83,30 @@ impl<T: bxcan::Instance> Can<T> {
         }
     }
 }
+
+mod private {
+    pub trait Sealed {}
+}
+
+pub trait WithInterrupt: bxcan::Instance + private::Sealed {
+    type TInstance;
+    type RInstance;
```

Comment:
add ` : Interrupt` trait bounds here

Reply:
Maybe name TxInterrupt/RxInterrupt? They're not peripheral instances, they're interrupts